src/vs/workbench/contrib/terminal/browser/xterm/suggestAddon.ts

```diff
@@ -399,12 +410,14 @@ export class SuggestAddon extends Disposable implements ITerminalAddon, ISuggest
 			return;
 		}
 		let handled = false;
+		let handledCursorDelta = 0;
 
 		// Backspace
 		if (data === '\x7f') {
 			if (this._additionalInput && this._additionalInput.length > 0 && this._cursorIndexDelta > 0) {
 				handled = true;
-				this._additionalInput = this._additionalInput.substring(0, this._cursorIndexDelta-- - 1) + this._additionalInput.substring(this._cursorIndexDelta);
+				this._additionalInput = this._additionalInput.substring(0, this._cursorIndexDelta - 1) + this._additionalInput.substring(this._cursorIndexDelta--);
```

Comment:
Opps 😱 

To be more safe for the future should we pull the decrement onto the next line?

```suggestion
				this._additionalInput = this._additionalInput.substring(0, this._cursorIndexDelta - 1) + this._additionalInput.substring(this._cursorIndexDelta);
				this._cursorIndexDelta--
```